### PR TITLE
Python 3.8 Controller Minimum

### PR DESCRIPTION
**Pull request by sivel**
##### SUMMARY
This PR bumps a few checks, changes some docs, and adds limited use of `pathlib` in the CLI stub to cause at least Py2 failures if someone regexes out the `SystemExit`

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
lib/ansible/cli/scripts/ansible_cli_stub.py

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/template/native_helpers.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 from ansible.module_utils._text import to_text
 from ansible.module_utils.common.collections import is_sequence, Mapping
 from ansible.module_utils.common.text.converters import container_to_text
-from ansible.module_utils.six import PY2, text_type, string_types
+from ansible.module_utils.six import text_type, string_types
```
Comment (mkrizek) [inline lib/ansible/template/native_helpers.py:18]: Do we plan to move away from using `six` in the controller code as well (in a separate PR)?

Comment (sivel) [reply lib/ansible/template/native_helpers.py:18]: I hadn't really given it much thought, but you are correct, it probably doesn't make much sense to use going forward.

Comment (webknjaz) [reply lib/ansible/template/native_helpers.py:18]: String/bytes helpers still make sense, though. At least for as long as we don't have type checking.

File: lib/ansible/plugins/callback/__init__.py [hunk 2]
```diff
@@ -22,26 +22,18 @@
 import difflib
 import json
 import sys
-
+from collections import OrderedDict
```
Comment (webknjaz) [inline lib/ansible/plugins/callback/__init__.py:25]: Nit: maybe keep the import section separator?
```suggestion

from collections import OrderedDict
```

**Review by webknjaz — APPROVED** [review]

File: lib/ansible/template/safe_eval.py [hunk 3]
```diff
@@ -80,47 +81,7 @@ def safe_eval(expr, locals=None, include_exceptions=False):
...
-            set(
-                (ast.Set,)
-            )
-        )
-
-    # And in Python 3.4 too
-    if sys.version_info[:2] >= (3, 4):
-        SAFE_NODES.update(
-            set(
-                (ast.NameConstant,)
```
Comment (s-hertel) [inline lib/ansible/template/safe_eval.py:95]: heh, added in 3.4 and deprecated already in 3.8

**Review by s-hertel — APPROVED** [review]

**Pull request merged**